### PR TITLE
feat(profile): basic functionality for profile pictures AP-476

### PR DESCRIPTION
**Pull request by josephmcg**
**What this PR does** 📖
- upload picture to textile on account creation, set hash in Solana and state
- render picture in various places
  - chat
  - quick profile
  - adding friend
  - friends list
  - status
  - profile settings
- general refactoring/formatting across files

**Which issue(s) this PR fixes** 🔨
AP-476

**Special notes for reviewers** 🗒️
- I believe textile is initialized twice during initial registration. There's a comment about moving this to a startup action
  - The pre-existing location (loadAccount in store/accounts/actions.ts)
  - In order to add image to bucket before solana picture registration (registerUser in store/accounts/actions.ts)
- You can only add a picture on account creation right now

**Additional comments** 🎤

### CODE REVIEW

File: components/interactables/QuickProfile/QuickProfile.vue [hunk 1]
```diff
@@ -2,8 +2,17 @@
 <script lang="ts">
 import Vue, { PropType } from 'vue'
 import { mapState } from 'vuex'
-import { User } from '~/types/ui/user'
 import { ArrowRightIcon } from 'satellite-lucide-icons'
+import { User } from '~/types/ui/user'
+
+declare module 'vue/types/vue' {
+  interface Vue {
+    text: string
```
Comment (WanderingHogan) [inline components/interactables/QuickProfile/QuickProfile.vue:10]: 💪 fewer typescript red squiggles is nice

File: components/views/navigation/sidebar/status/Status.html [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
     class="user-info has-tooltip has-tooltip-top has-tooltip-primary"
     v-on:click="$toast.show($t('ui.copied'))"
     :data-tooltip="$t('controls.copy_id')"
     v-clipboard:copy="accounts.active"
   >
-    <TypographyTitle 
-      :text="accounts.details.name" 
+    <TypographyTitle :text="accounts.details.name" :size="6" />
+    <TypographySubtitle
+      :text="accounts.details.status || 'No status set.'"
```
Comment (WanderingHogan) [inline components/views/navigation/sidebar/status/Status.html:14]: how will the translation service handle this

Comment (josephmcg) [reply components/views/navigation/sidebar/status/Status.html:14]: idk if it should be translated. Seems like that may encroach on the user's self expression. people can always google if theyre curious

File: components/interactables/QuickProfile/QuickProfile.vue (hunk 1 above)
Comment (josephmcg) [reply components/interactables/QuickProfile/QuickProfile.vue:10]: 🦕

File: components/views/navigation/sidebar/status/Status.html (hunk 2 above)
Comment (WanderingHogan) [reply components/views/navigation/sidebar/status/Status.html:14]: i specifically mean `No status set.`

**Review by stavares843 — CHANGES_REQUESTED** [review]
seems the photo is rendered on
- chat
- quick profile
- adding friend
- friends list
- status

Captura de ecrã 2022-01-20, às 00 44 34
Captura de ecrã 2022-01-20, às 00 45 15
Captura de ecrã 2022-01-20, às 00 45 22
Captura de ecrã 2022-01-20, às 00 45 32

but not on the profile

Captura de ecrã 2022-01-20, às 00 45 47

flow
- create account
- add photo
- go to profile

is that expected?

File: components/views/navigation/sidebar/status/Status.html (hunk 2 above)
Comment (josephmcg) [reply components/views/navigation/sidebar/status/Status.html:14]: ah gotcha, I'll fix that

**Comment by phillsatellite** [comment]
Screen Shot 2022-01-18 at 10 08 15 PM

I got this bug one time and couldn't replicate it again after clearing data once but I thought I should comment it still just in case we wanted to add an error message if an image doesn't properly load in for some reason

**Comment by josephmcg** [comment]
Thanks Sara, I thought I was missing one. should be fixed now.

I've seen that bug too Phil, thought I fixed it by adjusting the data being passed around. Let me know if you're able to recreate it after pulling these latest changes

**Review by WanderingHogan — APPROVED** [review]

**Review by stavares843 — APPROVED** [review]

**Pull request merged**